### ISSUE TITLE
Document running etcd3 under rkt

### ISSUE THREAD

**Issue by dghubble** [coreos/etcd#6177]
Add rkt-specific examples and documentation to [op-guide/container.md](https://github.com/coreos/etcd/blob/c68d60c99f25de19ecb7d84cc4a3433310a233a0/Documentation/op-guide/container.md) showing a 3-node cluster.

cc @gyuho

**Comment by dghubble** [comment]
I eventually got https://github.com/coreos/coreos-baremetal/pull/315/files to work. You might look at that to come up with a container example to save others from the trial and error here.

**Comment by xiang90** [comment]
@dghubble Can you send a pull request for rkt container in the doc as you are currently working on it?

**Comment by dghubble** [comment]
Sure, I'll get to that soon-ish.

**Comment by venezia** [comment]
I'm having problems running etcd within rkt (as a systemd unit) - etcd bails trying to validate the flags.

```rkt[5272]: 2016-08-24 17:03:00.700366 E | etcdmain: error verifying flags, '--name' is not a valid flag```

Similar unit in Docker works just fine

See attached unit files

[venezia-docker.txt](https://github.com/coreos/etcd/files/435346/venezia-docker.txt)
[venezia-rkt.txt](https://github.com/coreos/etcd/files/435345/venezia-rkt.txt)

**Comment by gyuho** [comment]
@venezia What etcd version are you running? Can you try specifying the binary path here?

```
         quay.io/coreos/etcd:v3.0.5 \
         -- --data-dir /ephemeral/etcd \
```

https://github.com/coreos/etcd/blob/master/Documentation/op-guide/container.md#docker

Try something like

```
sudo docker run --net=host --name etcd quay.io/coreos/etcd:${ETCD_VERSION} \
    /usr/local/bin/etcd \
...
```

**Comment by venezia** [comment]
Just to be clear Docker works just fine, rkt does not.  (you're crossing examples)

I did, at one point have ```--exec /usr/local/bin/rkt``` but got the same error

**Comment by dghubble** [comment]
I'll try to post the example this week, etcd3 runs in rkt in the example clusters.

**Comment by monder** [comment]
@venezia I have no problem running `etcd` in `rkt`

```
/usr/bin/rkt run \
 --dns=10.0.0.2 \
 --net=host \
 --volume volume-0,kind=host,source=/mnt/ebs/vol-69d3fdee/ \
 --mount volume=volume-0,target=/data-dir \
coreos.com/etcd:v3.0.6 \
 -- \
 --name etcd-n1 \
 --advertise-client-urls http://etcd-n1.k.fabu.lan:2379 \
 --initial-advertise-peer-urls http://etcd-n1.k.fabu.lan:2380 \
 --listen-client-urls http://0.0.0.0:2379 \
 --listen-peer-urls http://0.0.0.0:2380 \
 --initial-cluster etcd-n2=http://etcd-n2.k.fabu.lan:2380,etcd-n1=http://etcd-n1.k.fabu.lan:2380,etcd-n3=http://etcd-n3.k.fabu.lan:2380 \
 --initial-cluster-state existing \
---
```

The differences I see:
- I use the standard stage1
- You prefix each argument with `--`. I'm sure its not how it supose to work. You shoud start with `--`, pass all the arguments and end with `---` as per https://coreos.com/rkt/docs/latest/subcommands/run.html#passing-arguments

**Comment by venezia** [comment]
@monder You're right, however the error that I got back was a bit strange - I swore I saw something that said to do this.  Hard coding the DNS seems a bit rough, so i went with mount /etc/resolv.conf and we're good.

I went with
```bash
/usr/bin/rkt run --net=host \
         --stage1-path=/usr/lib/rkt/stage1-images/stage1-fly.aci \
         --insecure-options=image \
         --inherit-env=true \
         --volume ssl,kind=host,source=/etc/etcd/ssl,readOnly=true \
         --volume data,kind=host,source=/ephemeral/etcd,readOnly=false \
         --volume resolv-conf,kind=host,source=/etc/resolv.conf \
         --mount volume=ssl,target=/etc/etcd/ssl \
         --mount volume=data,target=/ephemeral \
         --mount volume=resolv-conf,target=/etc/resolv.conf \
         quay.io/coreos/etcd:v3.0.5 \
         -- --data-dir /ephemeral/etcd \
         --name etcd2379 \
         --peer-client-cert-auth \
         --peer-trusted-ca-file=/etc/etcd/ssl/peer-ca.pem \
         --peer-cert-file=/etc/etcd/ssl/peer.pem \
         --peer-key-file=/etc/etcd/ssl/peer-key.pem \
         --trusted-ca-file=/etc/etcd/ssl/server-ca.pem \
         --cert-file=/etc/etcd/ssl/server.pem \
         --key-file=/etc/etcd/ssl/server-key.pem \
         --client-cert-auth \
         --discovery-srv etcd2379.etcd.testcluster.io \
         --initial-advertise-peer-urls https://${DEFAULT_IPV4}:2380 \
         --initial-cluster-token myClusterToken \
         --initial-cluster-state new \
         --advertise-client-urls https://${DEFAULT_IPV4}:2379 \
         --listen-client-urls https://127.0.0.1:2379,https://127.0.0.1:4001,https://${DEFAULT_IPV4}:2379 \
         --listen-peer-urls https://127.0.0.1:2380,https://${DEFAULT_IPV4}:2380 \
         ---
```

**Comment by xiang90** [comment]
Fixed by #6274
